inclusive-design/wecount.inclusivedesign.ca
# feat: migrate resources for learn page from Airtable

## Description

This PR adds a new collection, `resources`, with a barebones Netlify CMS configuration for editing collection items. It also adds scripts to migrate existing resources from Airtable.

The script consists of the following:

1. Running `npm run migrate:airtable` will pull resources from the production Airtable base (specifically, a view named `Deduped` which pulls resources based on a `Duplicate` column).
2. These resources are added to an Eleventy collection named [`airtableResources`](https://github.com/greatislander/wecount.inclusivedesign.ca/blob/68ab7e351255460c17fc6c79598efb9cab51bb3e/.eleventy.js#L65-L67).
3. This collection is used to template Markdown files into `dist/airtable-resources/` via a [Nunjucks template](https://github.com/greatislander/wecount.inclusivedesign.ca/blob/68ab7e351255460c17fc6c79598efb9cab51bb3e/src/collections/resources.njk).
4. These Markdown files are [copied into `src/collections/resources`](https://github.com/greatislander/wecount.inclusivedesign.ca/blob/65df42ae134df5ba787801cb04856b0ae326f85c/package.json#L15), populating an Eleventy collection named [`resources`](https://github.com/greatislander/wecount.inclusivedesign.ca/blob/65df42ae134df5ba787801cb04856b0ae326f85c/.eleventy.js#L61-L65).

## Steps to test

1. Run `npm run migrate:airtable`.
2. Run `npm start`.
3. Test content editing in the CMS.

**Expected behavior:** Resources should appear in the `src/collections/resources/` directory and should be editable via the CMS.

## Additional information

A JSON file of Learn tags has been added to the global data directory. It will probably come in handy when implementing filtering.

## Related issues

- This is a first step towards #437.
- #494 should be merged first and this branch should be updated with those changes.
